lib/basecrm/services/deals_service.rb

```diff
@@ -105,6 +105,28 @@ def update(deal)
       Deal.new(root[:data])
     end
 
+    # Upsert a deal
+    #
+    # post '/deals/upsert?filter_param=filter_value 
+    #
+    # Create a new deal or update an existing, based on a value of a filter or a set of filters. 
+    # At least a single filter - query parameter - is required. If no parameters are present, the request will return an error.
+    # See full docs https://developers.getbase.com/docs/rest/reference/deals
+    #
+    # @param filters [Hash] - hash contain filters, one level deep e.g. { name: 'string', 'custom_fields[field]': 'value' }
+    # @param deal [Deal, Hash] - This object's attributes describe the object to be updated or created
+    # @return [Deal] The resulting object representing updated or created resource.
+    def upsert(filters, deal)
+      validate_upsert_filters!(filters)
+      validate_type!(deal)
+
+      attributes = sanitize(deal)
+      query_string = URI.encode_www_form(filters)
```

Comment:
i was trying to find a way on encoding the hash without using Rails only methods (like Object.to_query) or a new gem like Addressable, this seems to do the trick, but other opinions welcome